### PR TITLE
Feature request 1050. Fit image size

### PR DESCRIPTION
**Pull request by valeryvpetrov-dev**
# Description

- Feature request https://github.com/Swati4star/Images-to-PDF/issues/1050 . 
- Except standard document sizes: A4, A3, etc there is a new option "Fit size". Fit size allows to create pdf document with size of selected image.
- Caution! com.itextpdf.text.Document class is used while creating pdf file. It does not have API for making document with different page sizes (see Document constructors). So result document cannot be sized like here https://smallpdf.com/jpg-to-pdf ([see example screenshot](https://drive.google.com/file/d/1waC-Pjomovg_CDISwfmPoma8HlOmCdCe/view?usp=sharing)). So I decided to calculate common size for this case. Max width and max height are used to fit selected images so no one will be cropped.

# Demo
- Portrait-oriented image ([link to demo gif](https://drive.google.com/file/d/1xRyy_MCrYhAKMvWh8rnAGLc3sL4T6i5U/view?usp=sharing))
- Landscape-oriented image ([link to demo gif](https://drive.google.com/file/d/1hIjexgNbod7JZ6qvVAKi_QSjeX6jG51b/view?usp=sharing))
- Mixed-oriented images ([link to demo gif](https://drive.google.com/file/d/1c3UV4Gb0-CwlAhRG0PDN5sm3WVfI3V34/view?usp=sharing))

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

See demo block.

Please describe the tests that you ran to verify your changes.
- [x] `./gradlew assembleDebug assembleRelease`
- [x] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Review by codegsaini — COMMENTED** [review]
To fit images of the same size it is well enough 👍 
Can you try to solve the problem of mixed-size images also...

**Comment by valeryvpetrov-dev** [comment]
As I mention, com.itextpdf.text.Document does not provide an ability to create document with different page sizes. Document constructor accepts page size as required argument. Document default constructor sets A4 as a default value. Do you have any ideas how to escape this limitation?

**Comment by codegsaini** [comment]
> Do you have any ideas how to escape this limitation?

Engaged in some work, will look into this soon. ;)

**Comment by Swati4star** [comment]
Can you fix the merge conflicts?

**Comment by valeryvpetrov-dev** [comment]
> Can you fix the merge conflicts?

@Swati4star done

File: app/src/main/java/swati4star/createpdf/util/PageSizeUtils.java [hunk 1]
```diff
@@ -102,6 +110,26 @@ public MaterialDialog showPageSizeDialog(boolean saveValue) {
...
+    /**
+     * Calculates common page size for given images to fit all of them.
+     * - If images have different sizes than max width, max height is used to fit all of them.
+     * - If images have same size, than output returns exact size of image.
+     *
+     * @param imagesUri - input images URIs as Strings.
+     * @return common size of input images as Rectangle.
+     * @see com.itextpdf.text.Rectangle
+     */
+    public static Rectangle calculateCommonPageSize(List<String> imagesUri) {
```
[the remark this answers is not in the document]
Comment (valeryvpetrov-dev) [reply app/src/main/java/swati4star/createpdf/util/PageSizeUtils.java:122]: @codegsaini yes. More detailed description is available in comment

**Comment by Swati4star** [comment]
Thank you for the contribution! :)

**Pull request merged**